meshmode/discretization/poly_element.py

```diff
@@ -278,31 +262,23 @@ def _quadrature_rule(self):
         else:
             return mp.XiaoGimbutasSimplexQuadrature(self.order, dims)
 
-    @property
-    @memoize_method
-    def unit_nodes(self):
-        result = self._quadrature_rule().nodes
-        if len(result.shape) == 1:
-            result = np.array([result])
 
-        dim2, _ = result.shape
-        assert dim2 == self.mesh_el_group.dim
-
-        return result
-
-    @property
+class _MassMatrixQuadratureElementGroup(PolynomialSimplexElementGroupBase):
     @memoize_method
-    def weights(self):
-        return self._quadrature_rule().weights
-
+    def quadrature_rule(self):
+        basis_fcts = self.basis_obj().functions
+        nodes = self.interp_nodes
+        mass_matrix = mp.mass_matrix(basis_fcts, nodes)
+        weights = np.dot(mass_matrix,
+                         np.ones(len(basis_fcts)))
+        return mp.Quadrature(nodes, weights, exact_to=self.order)
 
-class _MassMatrixQuadratureElementGroup(PolynomialSimplexElementGroupBase):
     @property
-    @memoize_method
-    def weights(self):
-        return np.dot(
-                mass_matrix(self),
-                np.ones(len(self.basis_obj().functions)))
+    def interp_nodes(self):
```

Comment:
```suggestion
    def _interp_nodes(self):
```
(Since this is an interface internal to this class.)

Reply:
Done in https://github.com/inducer/meshmode/pull/164/commits/fe63f45d9df60c65b346a90648500148cf61e961!